decidim-debates/app/models/decidim/debates/debate.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Debates
+    # The data store for a Debate in the Decidim::Debates component. It stores a
+    # title, description and any other useful information to render a custom
+    # debate.
+    class Debate < Debates::ApplicationRecord
+      include Decidim::HasFeature
+      include Decidim::HasCategory
+      include Decidim::Resourceable
+      include Decidim::Followable
+      include Decidim::Comments::Commentable
+      include Decidim::HasScope
+      include Decidim::Authorable
+      include Decidim::Reportable
+      include Decidim::HasReference
+
+      feature_manifest_name "debates"
+
+      validates :title, presence: true
+
+      # Public: Checks whether the debate is official or not.
+      #
+      # Returns a boolean.
+      def official?
+        author.blank?
+      end
+
+      # Public: Overrides the `reported_content_url` Reportable concern method.
+      def reported_content_url
+        ResourceLocatorPresenter.new(self).url
+      end
+
+      # Public: Calculates whether the current debate is an AMA-styled one or not.
```

Comment:
Can you add a description of what an AMA debate is?

Reply:
AMA is the acronym for Ask Me Anything, a type of debate where a politician answers most voted citizens' questions about any subject for a limited time.
Therefore, an AMA debate must have a starting and end time.

Reply:
Done